### PR TITLE
feat(Core/Eluna): Add new hook for Eluna - OnLearnSpell

### PR DESCRIPTION
**Pull request by iThorgrim-Hub**
## Changes Proposed:
-  Adding New Eluna RegisterPlayerEvent

## Tests Performed:

```lua
local function onLearn(event, player, spell)
  print('ok '..spell)
end
RegisterPlayerEvent(40, onLearn)
```
image

## How to Test the Changes:

1. Just add new mod-eluna on your server and compile

### CODE REVIEW

**Review by Winfidonarleyan — APPROVED** [review]

**Review by Helias — APPROVED** [review]

**Comment by Helias** [comment]
We should keep all the available hooks in some documentation

cc @Yehonal

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -3204,6 +3204,9 @@ void Player::learnSpell(uint32 spellId)
         if (itr2 != m_spells.end() && itr2->second->State != PLAYERSPELL_REMOVED && !itr2->second->IsInSpec(m_activeSpec))
             learnSpell(itr2->first);
     }
+#ifdef ELUNA
+  sEluna->OnLearnSpell(this, spellId);
```
Comment (r-o-b-o-t-o) [inline src/server/game/Entities/Player/Player.cpp:3208]:
```suggestion
    sEluna->OnLearnSpell(this, spellId);
```

Comment (FrancescoBorzi) [reply src/server/game/Entities/Player/Player.cpp:3208]: @iThorgrim

**Comment by FrancescoBorzi** [comment]
@iThorgrim the build with Eluna is failing with:

```
src/server/game/Entities/Player/Player.cpp:3223:13: fatal error: no member named 'OnLearnSpell' in 'Eluna'
    sEluna->OnLearnSpell(this, spellId);
```

**Comment by r-o-b-o-t-o** [comment]
> @iThorgrim the build with Eluna is failing with:
> 
> ```
> src/server/game/Entities/Player/Player.cpp:3223:13: fatal error: no member named 'OnLearnSpell' in 'Eluna'
>     sEluna->OnLearnSpell(this, spellId);
> ```

It will build again when this one is merged in Eluna: https://github.com/ElunaLuaEngine/Eluna/pull/379

**Comment by FrancescoBorzi** [comment]
fair enough!

**Comment by FrancescoBorzi** [comment]
what's the status of this PR?

**Comment by acidmanifesto** [comment]
@iThorgrim what is the current status of this PR?

**Comment by Winfidonarleyan** [comment]
Need replace hook

**Comment by acidmanifesto** [comment]
Marked as abandon as there has been zero response from the pr maker in a long time.

**Pull request closed without merging**